main/modes/utilities/ScreenSaver/screensaver.c

```diff
@@ -0,0 +1,356 @@
+// Structs
+//==============================================================================
+
+typedef struct
+{
+    vec_t pos;         // Position and bounding box
+    vec_t velocity;    // current velocity
+    vec_t accumulator; // Velocity added
+    wsg_t image;       // Loaded WSG
+    bool isActive;     // Should draw
+} bouncingObject_t;
+
+typedef struct
+{
+    bouncingObject_t objs[ARRAY_SIZE(items)];
+    led_t leds[CONFIG_NUM_LEDS];
+    bool explosion;
+    int64_t explosionTimer;
+} screenSaverData_t;
+
+//==============================================================================
+// Variables
+//==============================================================================
+
+swadgeMode_t screenSaverMode = {
+    .modeName          = modeName,
+    .wifiMode          = NO_WIFI,
+    .overrideUsb       = false,
+    .usesAccelerometer = false,
+    .overrideSelectBtn = false,
+    .fnEnterMode       = screenEnterMode,
+    .fnExitMode        = screenExitMode,
+    .fnMainLoop        = screenMainLoop,
+};
+
+screenSaverData_t* ssd;
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+static void screenEnterMode(void)
+{
+    ssd = (screenSaverData_t*)heap_caps_calloc(1, sizeof(screenSaverData_t), MALLOC_CAP_8BIT);
+    for (int idx = 0; idx < ARRAY_SIZE(items); idx++)
+    {
+        loadWsg(items[idx], &ssd->objs[idx].image, true);
+        vec_t startPos = {
+            .x = (TFT_WIDTH - ssd->objs[idx].image.w) >> 1,
+            .y = (TFT_HEIGHT - ssd->objs[idx].image.h) >> 1,
+        };
+        ssd->objs[idx].pos        = startPos;
+        ssd->objs[idx].velocity.x = (esp_random() % MAX_VELOCITY) - (MAX_VELOCITY >> 1);
+        ssd->objs[idx].velocity.y = (esp_random() % MAX_VELOCITY) - (MAX_VELOCITY >> 1);
+    }
+
+    // Setup LEDs
+    for (int idx = 0; idx < CONFIG_NUM_LEDS; idx++)
+    {
+        led_t l        = {0};
+        ssd->leds[idx] = l;
+    }
+
+    // Set active images
+    ssd->objs[0].isActive = true; // Always active
+    if (trophyGetPoints(false, roboRunnerMode.modeName) == 1000)
+    {
+        ssd->objs[1].isActive = true;
+    }
+    if (trophyGetPoints(false, mainMenuMode.modeName) == 1000)
+    {
+        ssd->objs[2].isActive = true;
+    }
+}
+
+static void screenExitMode(void)
+{
+    for (int idx = 0; idx < ARRAY_SIZE(items); idx++)
+    {
+        freeWsg(&ssd->objs[idx].image);
+    }
+    heap_caps_free(ssd);
+}
+
+static void screenMainLoop(int64_t elapsedUs)
+{
+    // Handle input
+    buttonEvt_t evt;
+    while (checkButtonQueueWrapper(&evt))
+    {
+        // Allow user to back out of mode
+        if(evt.down)
+        {
+            ssd->explosion = true;
+            ssd->explosionTimer = 0;
+        }
+    }
+
+    // Update objects
+    updateObjects();
+
+    // Color explosion
+    if(ssd->explosion)
+    {
+        colorExplosion();
```

Comment:
Doing this once a frame is a little seizurey. Maybe consider calling `danceSmoothRainbow()` for some time with a small `arg` for a faster spin? The fast mode uses 4000 and the slow mode uses 20000